lib/rubocop/cop/style/unnecessary_arguments.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Checks for unnecessary arguments to builtin methods
+      #
+      # @example
+      #   # bad
+      #
+      #   array.join('')
+      #   [1, 2, 3].join("")
+      #   string.split(" ")
+      #   "first\nsecond".split(" ")
+      #
+      #   # good
+      #   array.join
+      #   [1, 2, 3].join
+      #   string.split
+      #   "first second".split
+      #
+      class UnnecessaryArguments < Cop
```

Comment:
For example, `RedundantSeparatorArgument` cop may be preferred for cop naming.
https://github.com/rubocop-hq/rubocop/issues/7077

Reply:
Since @bbatsov suggested extending this cop beyond builtin methods, naming this `RedundantArguments`.